### PR TITLE
983/remove assetsconfig from other pallets

### PR DESCRIPTION
**Pull request by HaidarJbeily7**
- Replaced usage of assets pallet public functions with AssetManager trait
- Removed assets::Config from the following pallets:
  - ceres-launchpad
  - ceres-token-locker
  - ceres-liquidity-locker
  - ceres-governance-platform
  - demeter-farming-platform
  - pool-xyk
  - price-tools 
  - technical
  - farming
  - vested-rewards
  - rewards
  - faucet
  - kensetsu
  - bridge-proxy
  - qa-tools
  - order-book
  - xst

### CODE REVIEW

**Review by wer1st — COMMENTED** [review]
You have to remove `assets` from all `Cargo.toml` files of pallets.
If you need `assets` dependency for tests / mock, please move it to `[dev-dependencies]`

**Comment by HaidarJbeily7** [comment]
I double checked all cargo.toml files and fixed where the issue of not removing assets dependency appears..

Also, I removed the assets config from the rest pallets:

- ceres-staking
- apollo-platform
- dex-api
- dex-manager
- hemes-governance-platform
- iroha-migration
- xor-fee
- referrals
- trading-pair
- pswap-distribution


FYI @wer1st

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: pallets/dex-manager/Cargo.toml [hunk 1]
```diff
@@ -46,6 +46,7 @@ std = [
     'codec/std',
     "scale-info/std",
     'serde/std',
+    'assets/std',
```
Comment (wer1st) [inline pallets/dex-manager/Cargo.toml:49]: delete

File: pallets/farming/Cargo.toml [hunk 2]
```diff
@@ -29,7 +29,7 @@ sp-core = { git = "https://github.com/sora-xor/substrate.git", branch = "polkado
 sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
Comment (wer1st) [inline pallets/farming/Cargo.toml:32]: delete empty line

File: pallets/pool-xyk/Cargo.toml [hunk 3]
```diff
@@ -30,7 +30,7 @@ sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polk
 sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 pallet-timestamp = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
Comment (wer1st) [inline pallets/pool-xyk/Cargo.toml:33]: delete

File: pallets/pool-xyk/src/action_deposit_liquidity.rs [hunk 4]
```diff
@@ -33,11 +33,11 @@ use frame_support::weights::Weight;
 use frame_support::{dispatch, ensure};
 use sp_runtime::traits::Zero;
 
-use common::prelude::{AssetInfoProvider, Balance, FixedWrapper};
-
 use crate::{to_balance, to_fixed_wrapper, AccountPools, PoolProviders, TotalIssuances};
 
-use crate::aliases::{AccountIdOf, AssetIdOf, TechAccountIdOf};
+use common::prelude::{AssetIdOf, AssetInfoProvider, Balance, FixedWrapper};
```
Comment (wer1st) [inline pallets/pool-xyk/src/action_deposit_liquidity.rs:38]: Move it to line 35

File: pallets/pool-xyk/src/action_poly_swap.rs [hunk 5]
```diff
@@ -32,11 +32,11 @@ use frame_support::dispatch;
 use frame_support::dispatch::DispatchResult;
 use frame_support::weights::Weight;
 
-use common::SwapRulesValidation;
-
-use crate::aliases::{AccountIdOf, AssetIdOf, DEXIdOf, PolySwapActionStructOf, TechAccountIdOf};
+use crate::aliases::{AccountIdOf, DEXIdOf, PolySwapActionStructOf, TechAccountIdOf};
 use crate::Config;
 
+use common::{AssetIdOf, SwapRulesValidation};
```
Comment (wer1st) [inline pallets/pool-xyk/src/action_poly_swap.rs:38]: Move it to line 34

File: pallets/technical/Cargo.toml [hunk 6]
```diff
@@ -29,7 +29,7 @@ sp-runtime = { git = "https://github.com/sora-xor/substrate.git", branch = "polk
 sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 twox-hash = { version = "1.5.0", default-features = false }
 
-assets = { path = "../assets", default-features = false }
+
```
Comment (wer1st) [inline pallets/technical/Cargo.toml:32]: delete

File: pallets/vested-rewards/Cargo.toml [hunk 7]
```diff
@@ -11,7 +11,7 @@ version = '1.2.1'
 targets = ['x86_64-unknown-linux-gnu']
 
 [dependencies]
-assets = { path = "../assets", default-features = false }
+
```
Comment (wer1st) [inline pallets/vested-rewards/Cargo.toml:14]: delete

File: pallets/xor-fee/Cargo.toml [hunk 8]
```diff
@@ -31,7 +31,7 @@ sp-std = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot
 sp-core = { git = "https://github.com/sora-xor/substrate.git", branch = "polkadot-v0.9.38", default-features = false }
 traits = { git = "https://github.com/open-web3-stack/open-runtime-module-library.git", package = "orml-traits", default-features = false }
 smallvec = "1.5.0"
-assets = { path = "../assets", default-features = false }
+
```
Comment (wer1st) [inline pallets/xor-fee/Cargo.toml:34]: delete

File: common/src/traits.rs [hunk 9]
```diff
@@ -1089,6 +1143,232 @@ impl<AssetId, AccountId, AssetSymbol, AssetName, BalancePrecision, ContentSource
...
+        name: AssetName,
+        precision: BalancePrecision,
+        initial_supply: Balance,
+        is_mintable: bool,
+        opt_content_src: Option<ContentSource>,
+        opt_desc: Option<Description>,
+    ) -> DispatchResult;
+
+    fn burn_from(
+        asset_id: Self::AssetId,
```
Comment (wer1st) [inline common/src/traits.rs:1210]:
```suggestion
        asset_id: &Self::AssetId,
```

File: common/src/traits.rs [hunk 10]
```diff
@@ -1089,6 +1143,232 @@ impl<AssetId, AccountId, AssetSymbol, AssetName, BalancePrecision, ContentSource
...
+
+    fn transfer_from(
+        asset_id: &Self::AssetId,
+        from: &T::AccountId,
+        to: &T::AccountId,
+        amount: Balance,
+    ) -> DispatchResult;
+
+    fn mint_to(
+        asset_id: Self::AssetId,
```
Comment (wer1st) [inline common/src/traits.rs:1224]:
```suggestion
        asset_id: &Self::AssetId,
```

**Pull request merged**